opennlp-tools/src/main/java/opennlp/tools/util/Sequence.java

```diff
@@ -122,7 +122,32 @@ public void add(String outcome, double p) {
    * @return Retrieves a list of outcomes for this {@link Sequence}.
    */
   public List<String> getOutcomes() {
-    return outcomes;
+    return List.copyOf(outcomes);
+  }
+
+  /**
+   * @return Retrieves the size of the outcomes for this {@link Sequence}.
+   */
+  public int getSize() {
+    return outcomes.size();
+  }
+
+  /**
+   * @param index must be greater than zero and must not exceed {@link Sequence#getSize()}.
```

Comment:
Reading it after the night, it should better read: "must be greater or equal to zero". 0 is an allowed value here. Will fix later.